app/controllers/admin/documents_controller.rb

```diff
@@ -5,8 +5,10 @@ class Admin::DocumentsController < AdminController
   def index; end
 
   def update
+    check_change_status
     if @document.send("#{params[:document][:status]}!")
```

Comment:
Bọc vào transaction nào chú
có bắt rescues exception nhé

enum sử method update! và nếu có lỗi nó sẽ raise excetion `ActiveRecord::RecordInvalid` đấy

`All of these have a bang counterpart (that is, save! and update!), which are stricter in that they raise the exception ActiveRecord::RecordInvalid if validation fails`

https://github.com/rails/rails/blob/b9ca94caea2ca6a6cc09abaffaad67b447134079/activerecord/lib/active_record/enum.rb#L151